### PR TITLE
Allow for validation set creation in the cifar setting

### PR DESCRIPTION
**Pull request by manuelhaussmann**
Optional creation of a separate validation set from the training data  for C10/C100

### CODE REVIEW

File: inferno/io/box/cifar.py [hunk 1]
```diff
@@ -2,12 +2,12 @@
...
 import torchvision.transforms as transforms
+from torch.utils.data.sampler import SubsetRandomSampler
 
 
-def get_cifar10_loaders(root_directory, train_batch_size=128, test_batch_size=100,
-                        download=False, augment=False):
-    # Data preparation for CIFAR10. Adapted from
-    # https://github.com/kuangliu/pytorch-cifar/blob/master/main.py
+def get_cifar10_loaders(root_directory, train_batch_size=128, test_batch_size=256,
+                        download=False, augment=False, validsize=None):
```
Comment (nasimrahaman) [inline inferno/io/box/cifar.py]: I think `validation_dataset_size` would be a bit more self-documenting.

**Comment by manuelhaussmann** [comment]
agreed
